sql/src/main/java/io/crate/operation/projectors/fetch/FetchProjector.java

```diff
@@ -394,4 +405,38 @@ protected void doRun() throws Exception {
             sendToDownstream(isLast, 0);
         }
     }
+
+    interface OutputRowResolver {
+
+        Row resolveOutputRow(Object[] inputCells);
+    }
+
+    private static class SingleOutputRowResolver implements OutputRowResolver {
+
+        private final Row outputRow;
+
+        SingleOutputRowResolver(Row outputRow) {
+            this.outputRow = outputRow;
+        }
+
+        @Override
+        public Row resolveOutputRow(Object[] inputCells) {
+            return outputRow;
+        }
+    }
+
+    private static class MultiOutputRowResolver implements OutputRowResolver {
+
+        private final List<Row> outputRows;
+
+        MultiOutputRowResolver(List<Row> outputRows) {
+            this.outputRows = outputRows;
+        }
+
+        @Override
+        public Row resolveOutputRow(Object[] inputCells) {
+            int relationInputIdx = outputRows.size() == 1 ? 0 : ((Byte) inputCells[0]).intValue();
```

Comment:
Seems like the size() == 1 case can't/shouldn't happen here since there is a check in the constructor which switches the OutputRowResolver? Or am I missing something?